aptos_sdk/asymmetric_crypto_wrapper.py

```diff
@@ -85,3 +90,133 @@ def deserialize(deserializer: Deserializer) -> Signature:
     def serialize(self, serializer: Serializer):
         serializer.uleb128(self.variant)
         serializer.struct(self.signature)
+
+
+class MultiPublicKey(asymmetric_crypto.PublicKey):
+    keys: List[PublicKey]
+    threshold: int
+
+    MIN_KEYS = 2
+    MAX_KEYS = 32
+    MIN_THRESHOLD = 1
+
+    def __init__(self, keys: List[asymmetric_crypto.PublicKey], threshold: int):
+        assert (
+            self.MIN_KEYS <= len(keys) <= self.MAX_KEYS
+        ), f"Must have between {self.MIN_KEYS} and {self.MAX_KEYS} keys."
+        assert (
+            self.MIN_THRESHOLD <= threshold <= len(keys)
+        ), f"Threshold must be between {self.MIN_THRESHOLD} and {len(keys)}."
+
+        # Ensure keys are wrapped
+        self.keys = []
+        for key in keys:
+            if isinstance(key, PublicKey):
+                self.keys.append(key)
+            else:
+                self.keys.append(PublicKey(key))
+
+        self.threshold = threshold
+
+    def __str__(self) -> str:
+        return f"{self.threshold}-of-{len(self.keys)} Multi key"
+
+    def verify(self, data: bytes, signature: asymmetric_crypto.Signature) -> bool:
+        try:
+            total_sig = cast(MultiSignature, signature)
+            assert self.threshold <= len(
+                total_sig.signatures
+            ), f"Insufficient signatures, {self.threshold} > {len(total_sig.signatures)}"
+
+            for idx, signature in total_sig.signatures:
+                assert (
+                    len(self.keys) > idx
+                ), f"Signature index exceeds available keys {len(self.keys)} < {idx}"
+                assert self.keys[idx].verify(
+                    data, signature
+                ), "Unable to verify signature"
+
+        except Exception:
```

Comment:
I think we shouldn't catch the exception here, otherwise the assert above will have no effect?